pkg/kubectl/cmd/delete.go

```diff
@@ -116,6 +116,7 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		SelectAllParam(deleteAll).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
 		Flatten().
+		Latest().
```

Comment:
Added in https://github.com/caesarxuchao/kubernetes/commit/feb8ba9c4eece62fb5ca67c5663023f0f5e0d603#diff-e9dcb8431a1536260288c4ec52d89dd4